h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1475,12 +1475,20 @@ public static Object javaSerializeReadPojo(byte [] bytes) {
     try {
       return new ObjectInputStream(new ByteArrayInputStream(bytes)).readObject();
     } catch (IOException e) {
-      String className = new String(bytes, 8, Math.min(40, Math.max(3, bytes[7])));
+      String className = nameOfClass(bytes);
       throw Log.throwErr(new RuntimeException("Failed to deserialize " + className, e));
     } catch (ClassNotFoundException e) {
       throw Log.throwErr(e);
     }
   }
+
+  static String nameOfClass(byte[] bytes) {
+    assert bytes != null : "pojo bytes can't be null";
```

Comment:
I can return Unknown; but actually this check should be happening in the main deserialization function, all the time. If bytes are bad, we should just say it. There's nothing there. I'll add it.
(The fact that assertion are off in production... Dijkstra ridiculed the idea, eons ago.)